services/pull/check.go

```diff
@@ -98,15 +98,19 @@ func CheckPullMergable(stdCtx context.Context, doer *user_model.User, perm *acce
 		}
 
 		if err := CheckPullBranchProtections(ctx, pr, false); err != nil {
-			if models.IsErrDisallowedToMerge(err) {
-				if force {
-					if isRepoAdmin, err2 := access_model.IsUserRepoAdmin(ctx, pr.BaseRepo, doer); err2 != nil {
-						return err2
-					} else if !isRepoAdmin {
-						return err
-					}
-				}
-			} else {
+			if !models.IsErrDisallowedToMerge(err) {
+				log.Error("Error whilst checking pull branch protection for %-v: %v", pr, err)
+				return err
+			}
+
+			if !force {
```

Comment:
Looks like the logic isn't equal to before?

Reply:
I think the new code is fine.

They do the same thing:

If the error is not "DisallowedToMerge": return the err (before: the `else` block)

If the err is `DisallowedToMerge`:  if not `force`, return the err (DisallowedToMerge)

If `force`, then check if the user is repo admin:

    * If they is admin, pass to next
    * If they isn't, return err (DisallowedToMerge)

Reply:
As far as I can see the original code didn't properly account for disallowed to merge and if force was false it just wouldn't pay attention to it